### PR TITLE
peaks is now an object with .mz and .intensities

### PR DESCRIPTION
**Pull request by jspaaks**
Refs #138 

- removed ``.mz`` and ``.intensities`` from ``Spectrum``
- added a ``Spikes`` class
    - ``Spikes`` has checks on length and type of ``.mz`` and ``.intensities``
    - ``Spikes`` has a check order of ``.mz``
    - ``Spikes`` does not support setting either ``.mz`` or ``.intensities`` separately for safety
    - ``Spikes`` instances can be unpacked like tuple
    - ``Spikes`` instances can be cloned
- updated everything that used ``spectrum.mz`` or ``spectrum.intensities``
- added a ``.peaks`` and a ``.losses`` of type ``Spikes`` to ``Spectrum``
- added tests for new class ``Spikes`` and updated existing tests

### CODE REVIEW

**Review by fdiblen — APPROVED** [review]
:heavy_check_mark:

**Comment by jspaaks** [comment]
@fdiblen Maybe the others should look at this too? I'll leave it open for a bit.

File: matchms/Spikes.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        assert intensities.dtype == "float", "Input argument 'intensities' should be an array of type float."
+
+        self._mz = mz
+        self._intensities = intensities
+
+        assert self._is_sorted(), "mz values are out of order."
+
+    def __eq__(self, other):
+        return \
+            numpy.allclose(self.mz, other.mz) and \
```
Comment (jspaaks) [inline matchms/Spikes.py:20]: should check the shape as well

File: matchms/Spectrum.py [hunk 2]
```diff
@@ -1,31 +1,32 @@
...
     def __eq__(self, other):
         return \
-            self.mz.shape == other.mz.shape and \
-            numpy.allclose(self.mz, other.mz) and \
-            self.intensities.shape == other.intensities.shape and \
-            numpy.allclose(self.intensities, other.intensities) and \
+            self.peaks.mz.shape == other.peaks.mz.shape and \
+            numpy.allclose(self.peaks.mz, other.peaks.mz) and \
+            self.peaks.intensities.shape == other.peaks.intensities.shape and \
+            numpy.allclose(self.peaks.intensities, other.peaks.intensities) and \
```
Comment (jspaaks) [inline matchms/Spectrum.py]: should delegate equality test to ``Spikes`` instance

Comment (jspaaks) [reply matchms/Spectrum.py]: should include ``losses`` as well

File: matchms/Spectrum.py [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
             self.metadata == other.metadata
 
     def clone(self):
         """Return a deepcopy of the spectrum instance."""
-        return Spectrum(mz=self.mz,
-                        intensities=self.intensities,
+        return Spectrum(mz=self.peaks.mz,
+                        intensities=self.peaks.intensities,
```
Comment (jspaaks) [inline matchms/Spectrum.py]: should include ``losses`` as well

File: matchms/Spectrum.py [hunk 4]
```diff
@@ -156,3 +137,13 @@ def metadata(self):
...
+
+    @property
+    def peaks(self):
+        """getter method for metadata private variable"""
+        return self._peaks.clone()
+
+    @peaks.setter
+    def peaks(self, value):
+        """setter method for metadata private variable"""
+        self._peaks = value
```
Comment (jspaaks) [inline matchms/Spectrum.py:159]: add getter and setter for ``losses``

File: matchms/Spikes.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            numpy.allclose(self.intensities, other.intensities)
+
+    def __getitem__(self, item):
+        return [self.mz, self.intensities].__getitem__(item)
+
```
Comment (jspaaks) [inline matchms/Spikes.py:29]: could probably do `` [self.mz, self.intensities][item]``?

File: matchms/Spikes.py (hunk 1 above)
Comment (jspaaks) [reply matchms/Spikes.py:20]: added

File: matchms/Spectrum.py (hunk 2 above)
Comment (jspaaks) [reply matchms/Spectrum.py]: added

File: matchms/Spectrum.py (hunk 3 above)
Comment (jspaaks) [reply matchms/Spectrum.py]: and metadata

Comment (jspaaks) [reply matchms/Spectrum.py]: added both

File: matchms/Spectrum.py (hunk 4 above)
Comment (jspaaks) [reply matchms/Spectrum.py:159]: added

File: matchms/Spikes.py (hunk 5 above)
Comment (jspaaks) [reply matchms/Spikes.py:29]: added

**Comment by fdiblen** [comment]
I suggest we merge this.

**Comment by jspaaks** [comment]
@fdiblen I agree but also I would like to get some feedback, particularly, Stefan's (he's in a call at the moment).

notify @sverhoeven

File: matchms/filtering/normalize_intensities.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import numpy
+from matchms import Spikes
```
Comment (sverhoeven) [inline matchms/filtering/normalize_intensities.py]: Either use relative imports every where or absolute. Here `from ..Spikes import Spikes`

Comment (sverhoeven) [reply matchms/filtering/normalize_intensities.py]: Created separate isssue https://github.com/matchms/matchms/issues/166 to fix import style.

**Review by sverhoeven — APPROVED** [review]
Reads much nicer and provides place for losses.

File: matchms/filtering/normalize_intensities.py (hunk 6 above)
Comment (jspaaks) [reply matchms/filtering/normalize_intensities.py]: made everything relative import

**Comment by sverhoeven** [comment]
Great that #166 is now also fixed.

**Pull request merged**